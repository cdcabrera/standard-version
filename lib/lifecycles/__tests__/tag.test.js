const tag = require('../tag');

jest.mock('../../run-execFile', () => (...args) => {
  console.info(args);
  return 'mock-exec-file-response';
});

describe('tag', () => {
  const mockInfo = jest.fn();
  let spyInfo;

  beforeEach(() => {
    spyInfo = jest.spyOn(console, 'info').mockImplementation(mockInfo);
  });

  afterEach(() => {
    mockInfo.mockClear();
    spyInfo.mockClear();
  });

  it('should skip returning tag information', async () => {
    await tag('lorem ipsum', undefined, { skip: { tag: true } });
    expect(mockInfo.mock.calls).toMatchSnapshot('console');
  });

  it('should run lifecycles returning tag information', async () => {
    await tag('lorem ipsum', undefined, {
      bumpFiles: [],
      skip: {},
      scripts: {
        prerelease: 'echo "prerelease lifecycle mock"',
        pretag: 'echo "pretag lifecycle mock"',
        posttag: 'echo "posttag lifecycle mock"'
      }
    });

    expect(mockInfo.mock.calls).toMatchSnapshot('console');
  });
});
