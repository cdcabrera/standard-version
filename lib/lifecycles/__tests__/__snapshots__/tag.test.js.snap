// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tag should run lifecycles returning tag information: console 1`] = `
Array [
  Array [
    "<green>✔</green> Running lifecycle script \\"<bold>pretag</bold>\\"",
  ],
  Array [
    "<blue>ℹ</blue> - execute command: \\"<bold>echo \\"pretag lifecycle mock\\"</bold>\\"",
  ],
  Array [
    "<green>✔</green> tagging release <bold>undefined</bold><bold>lorem ipsum</bold>",
  ],
  Array [
    Array [
      Object {
        "bumpFiles": Array [],
        "scripts": Object {
          "posttag": "echo \\"posttag lifecycle mock\\"",
          "prerelease": "echo \\"prerelease lifecycle mock\\"",
          "pretag": "echo \\"pretag lifecycle mock\\"",
        },
        "skip": Object {},
      },
      "git",
      Array [
        "tag",
        "-a",
        "undefinedlorem ipsum",
        "-m",
        "undefined",
      ],
    ],
  ],
  Array [
    Array [
      "",
      "git",
      Array [
        "rev-parse",
        "--abbrev-ref",
        "HEAD",
      ],
    ],
  ],
  Array [
    "<blue>ℹ</blue> Run \`<bold>git push --follow-tags origin mock-exec-file-response</bold>\` to publish",
  ],
  Array [
    "<green>✔</green> Running lifecycle script \\"<bold>posttag</bold>\\"",
  ],
  Array [
    "<blue>ℹ</blue> - execute command: \\"<bold>echo \\"posttag lifecycle mock\\"</bold>\\"",
  ],
]
`;

exports[`tag should skip returning tag information: console 1`] = `Array []`;
